dartsim/src/JointFeatures_TEST.cc

```diff
@@ -397,6 +418,134 @@ TEST_F(JointFeaturesFixture, LinkCountsInJointAttachDetach)
   EXPECT_EQ(0u, model2->GetLinkCount());
 }
 
+/////////////////////////////////////////////////
+// Attach a fixed joint between links that belong to different models where one
+// of the models is created after a step is called
+TEST_F(JointFeaturesFixture, JointAttachDetachSpawnedModel)
+{
+  std::string model1Str = R"(
+  <sdf version="1.6">
+    <model name="M1">
+      <pose>0 0 0.1 0 0 0</pose>
+      <link name="body">
+        <collision name="coll_box">
+          <geometry>
+            <box>
+              <size>0.2 0.2 0.2</size>
+            </box>
+          </geometry>
+        </collision>
+      </link>
+    </model>
+  </sdf>)";
+
+  std::string model2Str = R"(
+  <sdf version="1.6">
+    <model name="M2">
+      <pose>1 0 0.1 0 0 0</pose>
+      <link name="chassis">
+        <collision name="coll_sphere">
+          <geometry>
+            <sphere>
+              <radius>0.1</radius>
+            </sphere>
+          </geometry>
+        </collision>
+      </link>
+    </model>
+  </sdf>)";
+
+  physics::ForwardStep::Output output;
+  physics::ForwardStep::State state;
+  physics::ForwardStep::Input input;
+
+  physics::World3dPtr<TestFeatureList> world;
+  {
+    sdf::Root root;
+    const sdf::Errors errors = root.Load(TEST_WORLD_DIR "ground.sdf");
+    ASSERT_TRUE(errors.empty()) << errors.front();
+    world = this->engine->ConstructWorld(*root.WorldByIndex(0));
+    ASSERT_NE(nullptr, world);
+  }
+
+  {
+    sdf::Root root;
+    sdf::Errors errors = root.LoadSdfString(model1Str);
+    ASSERT_TRUE(errors.empty()) << errors.front();
+    ASSERT_NE(nullptr, root.ModelByIndex(0));
+    world->ConstructModel(*root.ModelByIndex(0));
+  }
+
+  world->Step(output, state, input);
+
+  {
+    sdf::Root root;
+    sdf::Errors errors = root.LoadSdfString(model2Str);
+    ASSERT_TRUE(errors.empty()) << errors.front();
+    ASSERT_NE(nullptr, root.ModelByIndex(0));
+    world->ConstructModel(*root.ModelByIndex(0));
+  }
+
+  const std::string modelName1{"M1"};
+  const std::string modelName2{"M2"};
+  const std::string bodyName1{"body"};
+  const std::string bodyName2{"chassis"};
+
+  auto model1 = world->GetModel(modelName1);
+  auto model2 = world->GetModel(modelName2);
+  auto model1Body = model1->GetLink(bodyName1);
+  auto model2Body = model2->GetLink(bodyName2);
+
+  dart::simulation::WorldPtr dartWorld = world->GetDartsimWorld();
+  ASSERT_NE(nullptr, dartWorld);
+
+  const auto skeleton1 = dartWorld->getSkeleton(modelName1);
+  const auto skeleton2 = dartWorld->getSkeleton(modelName2);
+  ASSERT_NE(nullptr, skeleton1);
+  ASSERT_NE(nullptr, skeleton2);
+
+  auto *dartBody1 = skeleton1->getBodyNode(bodyName1);
+  auto *dartBody2 = skeleton2->getBodyNode(bodyName2);
+
+  ASSERT_NE(nullptr, dartBody1);
+  ASSERT_NE(nullptr, dartBody2);
+
+  const auto poseParent = dartBody1->getTransform();
+  const auto poseChild = dartBody2->getTransform();
+
+  // Commenting out the following `step` call makes this test fail
```

Comment:
Is this comment accurate? If so, should the next line be uncommented?

Reply:
What I meant is that before this PR, commenting it out make it fail. Uncommenting it will make the test pass. I'll add a sentence to clarify.

Reply:
4458614